### ISSUE TITLE
Yara_draft is crashing starterator

### ISSUE THREAD

**Issue by cdshaffer** [cdshaffer/starterator#17]
only comment is crashes on 117th of 135 phams.

**Comment by cdshaffer** [comment]
Here is the traceback:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 551, in __bootstrap_inner
    self.run()
  File "/home/seastudent/PycharmProjects/starterator/starterator/uiStarterate.py", line 297, in run
    self.starterate()
  File "/home/seastudent/PycharmProjects/starterator/starterator/uiStarterate.py", line 302, in starterate
    gui=self, event=self.stop_thread)
  File "/home/seastudent/PycharmProjects/starterator/starterator/starterate.py", line 112, in starterate
    final_file, short_final = phage.final_report()
  File "/home/seastudent/PycharmProjects/starterator/starterator/report.py", line 52, in final_report
    self.make_reports()
  File "/home/seastudent/PycharmProjects/starterator/starterator/report.py", line 77, in make_reports
    pham = gene_report.make_report(whole=True)
  File "/home/seastudent/PycharmProjects/starterator/starterator/report.py", line 268, in make_report
    self.pham.find_most_common_start()
  File "/home/seastudent/PycharmProjects/starterator/starterator/phams.py", line 218, in find_most_common_start
    most_called_start_index = self.total_possible_starts.index(called_starts_count[0][0])+1
ValueError: 0 is not in list

**Comment by cdshaffer** [comment]
I tested the code on phage Dori, everything ran fine, was working with branch fixrevreseendgenes and yes Yara is crashing. So Yara is crashing even with the fix that stopped dori 94 from crashing. So I am pretty sure it is an unpatched bug. The gene is the first gene on reverse strand. Test out DNA Master autoannotation to get exact position of gene.

**Comment by cdshaffer** [comment]
Ok I added code to phage.py that checks the gene to see if the length of the gene as reported from the database is the same as what you get if you calculate the difference between the start and the stop. If there is a discrepancy then that gene and its pham are NOT added to the dictionary of phams to run through the starterator analysis. Checked Yara with this code update and no crash, also tested Dori and no new problems. The branch is called filterSpanningGenes, it has two updates, one is to include length from the gene table, then check if stop - start == length. If this test fails it will skip that gene.
